common/app/templates/inlineJS/blocking/shouldEnhance.scala.js

```diff
@@ -35,7 +46,7 @@
         return (navigator.platform === 'iPad');
     };
 
-    window.shouldEnhance = !personPrefersCore() && !isOlderIOSDevice() && !(@item.isFront && isIpad());
+    window.shouldEnhance = personDemandsFeatures() || (!personPrefersCore() && !isOlderIOSDevice() && !(@item.isFront && isIpad()));
```

Comment:
Could this be simpler? We could have a function, `isCore`, that returns true or false. Then this becomes:

``` js
window.shouldEnhance = !isCore() || (!isCore() && !isOlderIOSDevice() || !(@item.isFront && isIpad()))
```